platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r25/TestProgressCrossVersionSpec.groovy

```diff
@@ -163,25 +165,26 @@ class TestProgressCrossVersionSpec extends ToolingApiSpecification implements Wi
         workerSuite.descriptor.methodName == null
         workerSuite.descriptor.parent == rootSuite.descriptor
 
-        def testClass = events.operation("Test class example.MyTest")
+        def testClass = events.operation("MyTest")
         testClass.descriptor.jvmTestKind == JvmTestKind.SUITE
         testClass.descriptor.name == 'example.MyTest'
-        testClass.descriptor.displayName == 'Test class example.MyTest'
+        testClass.descriptor.displayName == 'MyTest'
```

Comment:
This change breaks the contract of `OperationDescriptor.displayName`, which is to supply enough information for a human to identify the operation in isolation.

Instead, I would add a new property to the test operation descriptor that provides a contextual display name that a human can interpret relative to the parent operation. The IDE can then decide which of these to use based on how the operation is displayed in the UI.

Reply:
> This change breaks the contract of OperationDescriptor.displayName, which is to supply enough information for a human to identify the operation in isolation.

Can you provide some examples when it does matter? The [current JavaDoc](https://github.com/gradle/gradle/blob/92c817150c7597513ad7ec7147a118c27c5636fb/platforms/ide/tooling-api/src/main/java/org/gradle/tooling/events/OperationDescriptor.java#L43) does not imply "in isolation". 
And with Junit5 test an `@DisplayName`, I don't think this contract is held in the current version of Gradle.

Apart from unit tests, I've fixed some integration tests and the only place I found where it would make sense to preserve the old displayName is the detailed message [here](https://github.com/gradle/gradle/blob/48ab946ecf65e2560f85915c63842185b7834246/platforms/ide/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/TestExecutionResultEvaluator.java#L126). But for that particular case, it can be solved in a different way.